dynamic_rest/serializers.py

```diff
@@ -147,6 +149,18 @@ def __new__(cls, *args, **kwargs):
         if not issubclass(list_serializer_class, DynamicListSerializer):
             list_serializer_class = DynamicListSerializer
         meta.list_serializer_class = list_serializer_class
+
+        # If a lookup field has been defined in the serializer that
+        # field becomes the ID field for the serializer
+        lookup_field = getattr(meta, 'lookup_field', None)
+        if lookup_field is not None:
+            model = getattr(cls.Meta, 'model')
+            info = model_meta.get_field_info(model)
+            # XXX: This might not be the way to correctly solve this
+            cls._declared_fields['id'] = \
```

Comment:
It is done normally by metaclass https://github.com/encode/django-rest-framework/blob/a8e527a4625b48914a553256f8be2907bfcc7c99/rest_framework/serializers.py#L288 we want to replace the id field of the serializer runtime for this to work. I'll look again at this tomorrow with a fresh perspective, if in the meantime you have any additional thoughts on this.